# Supply Chain Analysis SQLite

This project analyzes supply chain data, particularly focusing on energy consumption, carbon emissions, renewable energy use, and environmental impact among companies.

### Objectives

The project is likely aimed at analyzing the environmental sustainability of companies within a supply chain context. This could involve:

1. **Energy Consumption Analysis**: Identifying the companies with the highest energy consumption (in MWh), which is a key factor in understanding a company’s environmental footprint. The project is aimed at identifying which companies are using the most energy, potentially leading to insights about energy inefficiencies or opportunities for improvement.

2. **Carbon Emissions**: Analyzing the relationship between carbon emissions and energy use, especially looking into how companies’ use of renewable energy might influence their overall carbon emissions. This can help identify companies that are more eco-friendly or those that need to take steps to reduce their carbon footprint.

3. **Environmental Impact Scores**: Focusing on environmental impact scores (i.e., companies with the highest environmental impact), which is a key measure of sustainability efforts. These scores may take into account factors like energy consumption, carbon emissions, and other environmental concerns, and identifying companies with the highest impact can help target areas for improvement.

4. **Use of Renewable Energy**: Exploring how companies are using renewable energy and its relationship with their carbon emissions. This is crucial for assessing whether companies are integrating more sustainable energy sources into their operations.

5. **Customer Satisfaction and Supply Chain Agility**: Investigating how customer satisfaction correlates with supply chain agility (i.e., how quickly and efficiently a company can adapt its supply chain to changes). A more agile supply chain could potentially lead to better customer satisfaction by responding faster to market demands.

### **Dataset**

The dataset titled "SCM Dataset with Green Logistics" comprises a comprehensive collection of data from various companies, focusing on their supply chain management (SCM) practices, performance metrics, and green logistics initiatives. Key features of the dataset include:

- **SCM Practices**: Information on various supply chain management practices, such as Agile SCM, Lean Manufacturing, and Cross-Docking.
- **Performance Metrics**: Quantitative metrics like Supplier Count, Inventory Turnover Ratio, Lead Time (days), Order Fulfillment Rate (%), and Customer Satisfaction (%).
- **Technology Utilized**: Details on technologies used by companies, such as ERP, AI, Robotics, and Blockchain.
- **Environmental Impact Metrics**: Data on carbon emissions (kg CO2e), recycling rates (%), energy consumption (MWh), use of renewable energy (%), and green packaging usage (%).
- **Sustainability Indicators**: Environmental Impact Score, Supply Chain Agility, Sustainability Practices, and Supply Chain Complexity Index.
- **Financial Metrics**: Cost of Goods Sold (COGS), Operational Efficiency Score, Revenue Growth Rate, and Supply Chain Risk (%).
- **Collaborative Metrics**: Supplier Collaboration Level, Supply Chain Resilience Score, and Supplier Relationship Score.

### 1. Import Libraries

In [1]:
import pandas as pd
import sqlite3

### 2. Load the Excel File

In [3]:
# Load your Excel file into a DataFrame
file_path = 'SCM_Dataset_Updated_with_Green_Logistics.xlsx'
df = pd.read_excel(file_path)

# Display the first few rows to check the data
df.head()

Company Name       SCM Practices  Supplier Count  Inventory Turnover Ratio  \
0        Apple           Agile SCM           300.0                       7.0   
1    Microsoft  Lean Manufacturing           200.0                       5.2   
2       Google       Cross-Docking         20000.0                       7.5   
3       Amazon           Agile SCM           500.0                      10.5   
4     Facebook  Lean Manufacturing           100.0                       5.8   

   Lead Time (days)  Order Fulfillment Rate (%)  Customer Satisfaction (%)  \
0               8.0                        94.0                       91.0   
1              15.0                        91.0                       89.0   
2               4.0                        95.0                       90.0   
3              15.0                        95.0                       90.0   
4              12.0                        89.0                       88.0   

   Technology Utilized  Environmental Impact Score Supply Chain Agility  ...  \
0  ERP, AI, Blockchain                        70.0                 High  ...   
1    ERP, AI, Robotics                        75.0                 High  ...   
2  AI, ERP, Blockchain                        70.0                 High  ...   
3  ERP, AI, Blockchain                        70.0                 High  ...   
4    ERP, AI, Robotics                        75.0                 High  ...   

   Supply Chain Risk (%)  Supplier Collaboration Level  \
0                    7.0                          High   
1                   12.0                        Medium   
2                    5.0                          High   
3                   12.0                        Medium   
4                   10.0                        Medium   

   Supply Chain Resilience Score Supplier Relationship Score  \
0                           90.0                        85.0   
1                           85.0                        85.0   
2                           90.0                        85.0   
3                           88.0                        85.0   
4                           85.0                        80.0   

  Total Implementation Cost Carbon Emissions (kg CO2e) Recycling Rate (%)  \
0                  48566365                     120000                 60   
1                  54456148                     121000                 61   
2                  78733131                     122000                 62   
3                   1635157                     123000                 63   
4                  73150360                     124000                 64   

   Energy Consumption (MWh)  Use of Renewable Energy (%)  \
0                    100000                           50   
1                    100500                           51   
2                    101000                           52   
3                    101500                           53   
4                    102000                           54   

   Green Packaging Usage (%)  
0                         40  
1                         41  
2                         42  
3                         43  
4                         44  

[5 rows x 29 columns]

### 3. Create an SQLite Database

In [4]:
# Connect to an SQLite database (or create it if it doesn't exist)
conn = sqlite3.connect('scm_data.db')
cursor = conn.cursor()

# Create a table (if needed) and insert the DataFrame data
df.to_sql('scm_table', conn, if_exists='replace', index=False)

1000

### 4. Querying the SQLite Database

#### 1. List of Companies and Their SCM Practices

This query provides a list of companies and their supply chain practices.

In [5]:
# Create an in-memory SQLite database
conn = sqlite3.connect(':memory:')

# Sample data (based on your provided sample)
data = {
    'Company_Name': ['Apple', 'Microsoft', 'Google', 'Amazon', 'Facebook'],
    'SCM_Practices': ['Agile SCM', 'Lean Manufacturing', 'Cross-Docking', 'Agile SCM', 'Lean Manufacturing'],
    'Supplier_Count': [300.0, 200.0, 20000.0, 500.0, 100.0],
    'Inventory_Turnover_Ratio': [7.0, 5.2, 7.5, 10.5, 5.8],
    'Lead_Time_days': [8.0, 15.0, 4.0, 15.0, 12.0],
    'Order_Fulfillment_Rate': [94.0, 91.0, 95.0, 95.0, 89.0],
    'Customer_Satisfaction': [91.0, 89.0, 90.0, 90.0, 88.0],
    'Technology_Utilized': ['ERP, AI, Blockchain', 'ERP, AI, Robotics', 'AI, ERP, Blockchain', 'ERP, AI, Blockchain', 'ERP, AI, Robotics'],
    'Environmental_Impact_Score': [70.0, 75.0, 70.0, 70.0, 75.0],
    'Supply_Chain_Agility': ['High', 'High', 'High', 'High', 'High'],
    'Supply_Chain_Risk': [7.0, 12.0, 5.0, 12.0, 10.0],
    'Supplier_Collaboration_Level': ['High', 'Medium', 'High', 'Medium', 'Medium'],
    'Supply_Chain_Resilience_Score': [90.0, 85.0, 90.0, 88.0, 85.0],
    'Supplier_Relationship_Score': [85.0, 85.0, 85.0, 85.0, 80.0],
    'Total_Implementation_Cost': [48566365, 54456148, 78733131, 1635157, 73150360],
    'Carbon_Emissions_kg_CO2e': [120000, 121000, 122000, 123000, 124000],
    'Recycling_Rate': [60, 61, 62, 63, 64],
    'Energy_Consumption_MWh': [100000, 100500, 101000, 101500, 102000],
    'Use_of_Renewable_Energy': [50, 51, 52, 53, 54],
    'Green_Packaging_Usage': [40, 41, 42, 43, 44]
}

# Create a dataframe
df = pd.DataFrame(data)

# Load the dataframe into the SQLite database
df.to_sql('supply_chain', conn, index=False, if_exists='replace')

# Query to list companies and their SCM practices
query = '''
SELECT Company_Name, SCM_Practices 
FROM supply_chain
ORDER BY Company_Name
'''

# Execute the query and fetch the results
result = pd.read_sql_query(query, conn)

# Display the results
print(result)

  Company_Name       SCM_Practices
0       Amazon           Agile SCM
1        Apple           Agile SCM
2     Facebook  Lean Manufacturing
3       Google       Cross-Docking
4    Microsoft  Lean Manufacturing


#### 2. Find Companies with High Customer Satisfaction

This query identifies the companies that have high customer satisfaction (you can define "high" as over 90%).

In [6]:
# Query to find companies with high customer satisfaction (>90%)
query = '''
SELECT Company_Name, Customer_Satisfaction, SCM_Practices, Order_Fulfillment_Rate
FROM supply_chain
WHERE Customer_Satisfaction > 90
ORDER BY Customer_Satisfaction DESC
'''

# Execute the query and fetch the results
result = pd.read_sql_query(query, conn)

# Display the results
print("Companies with High Customer Satisfaction (>90%):")
print(result)

Companies with High Customer Satisfaction (>90%):
  Company_Name  Customer_Satisfaction SCM_Practices  Order_Fulfillment_Rate
0        Apple                   91.0     Agile SCM                    94.0


#### 3. Find Companies Using Advanced Technologies (AI, Blockchain, Robotics)

This query lists companies that are using AI, Blockchain, or Robotics in their supply chain operations.

In [7]:
# Query to find companies using advanced technologies
query = '''
SELECT 
    Company_Name, 
    Technology_Utilized,
    SCM_Practices,
    CASE 
        WHEN Technology_Utilized LIKE '%AI%' THEN 'Yes' 
        ELSE 'No' 
    END AS Uses_AI,
    CASE 
        WHEN Technology_Utilized LIKE '%Blockchain%' THEN 'Yes' 
        ELSE 'No' 
    END AS Uses_Blockchain,
    CASE 
        WHEN Technology_Utilized LIKE '%Robotics%' THEN 'Yes' 
        ELSE 'No' 
    END AS Uses_Robotics
FROM supply_chain
WHERE 
    Technology_Utilized LIKE '%AI%' OR 
    Technology_Utilized LIKE '%Blockchain%' OR 
    Technology_Utilized LIKE '%Robotics%'
ORDER BY Company_Name
'''

# Execute the query and fetch the results
result = pd.read_sql_query(query, conn)

# Display the results
print("Companies Using Advanced Technologies (AI, Blockchain, Robotics):")
print(result)

Companies Using Advanced Technologies (AI, Blockchain, Robotics):
  Company_Name  Technology_Utilized       SCM_Practices Uses_AI  \
0       Amazon  ERP, AI, Blockchain           Agile SCM     Yes   
1        Apple  ERP, AI, Blockchain           Agile SCM     Yes   
2     Facebook    ERP, AI, Robotics  Lean Manufacturing     Yes   
3       Google  AI, ERP, Blockchain       Cross-Docking     Yes   
4    Microsoft    ERP, AI, Robotics  Lean Manufacturing     Yes   

  Uses_Blockchain Uses_Robotics  
0             Yes            No  
1             Yes            No  
2              No           Yes  
3             Yes            No  
4              No           Yes  


#### 4. Top Companies with Highest Supply Chain Agility

This query returns the companies with the highest supply chain agility (I can adjust the definition of high agility as needed).

In [8]:
# Query to find companies with highest supply chain agility
# We'll rank companies with "High" agility by using lead time (lower is better) 
# and inventory turnover ratio (higher is better) as tiebreakers
query = '''
SELECT 
    Company_Name, 
    Supply_Chain_Agility,
    SCM_Practices,
    Lead_Time_days,
    Inventory_Turnover_Ratio,
    Order_Fulfillment_Rate,
    Supply_Chain_Resilience_Score
FROM supply_chain
WHERE Supply_Chain_Agility = 'High'
ORDER BY 
    Lead_Time_days ASC,
    Inventory_Turnover_Ratio DESC,
    Order_Fulfillment_Rate DESC
LIMIT 5
'''

# Execute the query and fetch the results
result = pd.read_sql_query(query, conn)

# Display the results
print("Top Companies with Highest Supply Chain Agility:")
print(result)

Top Companies with Highest Supply Chain Agility:
  Company_Name Supply_Chain_Agility       SCM_Practices  Lead_Time_days  \
0       Google                 High       Cross-Docking             4.0   
1        Apple                 High           Agile SCM             8.0   
2     Facebook                 High  Lean Manufacturing            12.0   
3       Amazon                 High           Agile SCM            15.0   
4    Microsoft                 High  Lean Manufacturing            15.0   

   Inventory_Turnover_Ratio  Order_Fulfillment_Rate  \
0                       7.5                    95.0   
1                       7.0                    94.0   
2                       5.8                    89.0   
3                      10.5                    95.0   
4                       5.2                    91.0   

   Supply_Chain_Resilience_Score  
0                           90.0  
1                           90.0  
2                           85.0  
3                           8

#### 5. Explore Lead Time and Customer Satisfaction

This query explores the relationship between lead time and customer satisfaction for each company.

In [10]:
# Query to explore the relationship between lead time and customer satisfaction
query = '''
SELECT 
    Company_Name, 
    Lead_Time_days,
    Customer_Satisfaction,
    SCM_Practices,
    Order_Fulfillment_Rate,
    CASE 
        WHEN Lead_Time_days <= 5 THEN 'Very Fast (≤5 days)'
        WHEN Lead_Time_days <= 10 THEN 'Fast (6-10 days)'
        WHEN Lead_Time_days <= 15 THEN 'Moderate (11-15 days)'
        ELSE 'Slow (>15 days)'
    END AS Lead_Time_Category
FROM supply_chain
ORDER BY Lead_Time_days
'''

# Execute the query and fetch the results
result = pd.read_sql_query(query, conn)

# Display the results
print("Relationship Between Lead Time and Customer Satisfaction:")
print(result)

# Calculate correlation coefficient using Python/pandas instead of SQLite
# Extract the relevant columns for correlation calculation
lead_time_satisfaction_query = '''
SELECT Lead_Time_days, Customer_Satisfaction 
FROM supply_chain
'''
corr_data = pd.read_sql_query(lead_time_satisfaction_query, conn)

# Calculate correlation using pandas
correlation = corr_data['Lead_Time_days'].corr(corr_data['Customer_Satisfaction'])
print("\nCorrelation Coefficient between Lead Time and Customer Satisfaction:")
print(round(correlation, 4))

# Calculate average customer satisfaction by lead time category
avg_by_category_query = '''
SELECT 
    CASE 
        WHEN Lead_Time_days <= 5 THEN 'Very Fast (≤5 days)'
        WHEN Lead_Time_days <= 10 THEN 'Fast (6-10 days)'
        WHEN Lead_Time_days <= 15 THEN 'Moderate (11-15 days)'
        ELSE 'Slow (>15 days)'
    END AS Lead_Time_Category,
    ROUND(AVG(Customer_Satisfaction), 2) AS Avg_Customer_Satisfaction,
    COUNT(*) AS Number_of_Companies
FROM supply_chain
GROUP BY Lead_Time_Category
ORDER BY MIN(Lead_Time_days)
'''

avg_by_category = pd.read_sql_query(avg_by_category_query, conn)
print("\nAverage Customer Satisfaction by Lead Time Category:")
print(avg_by_category)

Relationship Between Lead Time and Customer Satisfaction:
  Company_Name  Lead_Time_days  Customer_Satisfaction       SCM_Practices  \
0       Google             4.0                   90.0       Cross-Docking   
1        Apple             8.0                   91.0           Agile SCM   
2     Facebook            12.0                   88.0  Lean Manufacturing   
3    Microsoft            15.0                   89.0  Lean Manufacturing   
4       Amazon            15.0                   90.0           Agile SCM   

   Order_Fulfillment_Rate     Lead_Time_Category  
0                    95.0    Very Fast (≤5 days)  
1                    94.0       Fast (6-10 days)  
2                    89.0  Moderate (11-15 days)  
3                    91.0  Moderate (11-15 days)  
4                    95.0  Moderate (11-15 days)  

Correlation Coefficient between Lead Time and Customer Satisfaction:
-0.4326

Average Customer Satisfaction by Lead Time Category:
      Lead_Time_Category  Avg_Customer_Sa

#### 6. Companies with Low Carbon Emissions and High Green Packaging Usage

This query looks for companies with low carbon emissions and high green packaging usage, which may reflect their commitment to sustainability.

In [11]:
# Find median values for carbon emissions and green packaging usage
# This will help us define what's "low" and "high"
stats_query = '''
SELECT 
    AVG(Carbon_Emissions_kg_CO2e) AS Avg_Carbon_Emissions,
    MIN(Carbon_Emissions_kg_CO2e) AS Min_Carbon_Emissions,
    MAX(Carbon_Emissions_kg_CO2e) AS Max_Carbon_Emissions,
    AVG(Green_Packaging_Usage) AS Avg_Green_Packaging,
    MIN(Green_Packaging_Usage) AS Min_Green_Packaging,
    MAX(Green_Packaging_Usage) AS Max_Green_Packaging
FROM supply_chain
'''

stats = pd.read_sql_query(stats_query, conn)
print("Statistics for Sustainability Metrics:")
print(stats)

# Query to find companies with low carbon emissions and high green packaging
# We'll define "low" carbon emissions as below average
# And "high" green packaging as above average
query = '''
SELECT 
    Company_Name,
    Carbon_Emissions_kg_CO2e,
    Green_Packaging_Usage,
    Recycling_Rate,
    Use_of_Renewable_Energy,
    Environmental_Impact_Score,
    (
        CASE 
            WHEN Carbon_Emissions_kg_CO2e < (SELECT AVG(Carbon_Emissions_kg_CO2e) FROM supply_chain) 
            THEN 'Low' 
            ELSE 'High' 
        END
    ) AS Carbon_Emission_Level,
    (
        CASE 
            WHEN Green_Packaging_Usage > (SELECT AVG(Green_Packaging_Usage) FROM supply_chain) 
            THEN 'High' 
            ELSE 'Low' 
        END
    ) AS Green_Packaging_Level
FROM supply_chain
WHERE 
    Carbon_Emissions_kg_CO2e < (SELECT AVG(Carbon_Emissions_kg_CO2e) FROM supply_chain) 
    AND 
    Green_Packaging_Usage > (SELECT AVG(Green_Packaging_Usage) FROM supply_chain)
ORDER BY 
    Carbon_Emissions_kg_CO2e ASC,
    Green_Packaging_Usage DESC
'''

# Execute the query and fetch the results
result = pd.read_sql_query(query, conn)

# Display the results
print("\nCompanies with Low Carbon Emissions and High Green Packaging Usage:")
print(result)

# Alternative query with sustainability rank
rank_query = '''
SELECT 
    Company_Name,
    Carbon_Emissions_kg_CO2e,
    Green_Packaging_Usage,
    Recycling_Rate,
    Use_of_Renewable_Energy,
    Environmental_Impact_Score,
    (Carbon_Emissions_kg_CO2e - (SELECT MIN(Carbon_Emissions_kg_CO2e) FROM supply_chain)) / 
        ((SELECT MAX(Carbon_Emissions_kg_CO2e) FROM supply_chain) - (SELECT MIN(Carbon_Emissions_kg_CO2e) FROM supply_chain)) 
        AS Normalized_Carbon_Emissions,
    (Green_Packaging_Usage - (SELECT MIN(Green_Packaging_Usage) FROM supply_chain)) / 
        ((SELECT MAX(Green_Packaging_Usage) FROM supply_chain) - (SELECT MIN(Green_Packaging_Usage) FROM supply_chain)) 
        AS Normalized_Green_Packaging
FROM supply_chain
ORDER BY 
    Normalized_Carbon_Emissions ASC,
    Normalized_Green_Packaging DESC
'''

rank_result = pd.read_sql_query(rank_query, conn)

# Calculate sustainability score (lower is better)
# Normalized Carbon Emissions (lower is better) - weight: 60%
# Normalized Green Packaging (higher is better) - weight: 40%
rank_result['Sustainability_Score'] = (0.6 * rank_result['Normalized_Carbon_Emissions']) + (0.6 * (1 - rank_result['Normalized_Green_Packaging']))
rank_result = rank_result.sort_values(by='Sustainability_Score')

print("\nCompanies Ranked by Overall Sustainability Score (lower is better):")
# Format to 2 decimal places for readability
pd.set_option('display.float_format', '{:.2f}'.format)
print(rank_result[['Company_Name', 'Carbon_Emissions_kg_CO2e', 'Green_Packaging_Usage', 'Sustainability_Score']])

Statistics for Sustainability Metrics:
   Avg_Carbon_Emissions  Min_Carbon_Emissions  Max_Carbon_Emissions  \
0              122000.0                120000                124000   

   Avg_Green_Packaging  Min_Green_Packaging  Max_Green_Packaging  
0                 42.0                   40                   44  

Companies with Low Carbon Emissions and High Green Packaging Usage:
Empty DataFrame
Columns: [Company_Name, Carbon_Emissions_kg_CO2e, Green_Packaging_Usage, Recycling_Rate, Use_of_Renewable_Energy, Environmental_Impact_Score, Carbon_Emission_Level, Green_Packaging_Level]
Index: []

Companies Ranked by Overall Sustainability Score (lower is better):
  Company_Name  Carbon_Emissions_kg_CO2e  Green_Packaging_Usage  \
0        Apple                    120000                     40   
1    Microsoft                    121000                     41   
2       Google                    122000                     42   
3       Amazon                    123000                     43 

#### 7. Find Companies with Highest Energy Consumption

This query shows the companies with the highest energy consumption.

In [32]:
# Query to find top 10 companies with highest energy consumption
query = '''
SELECT 
    Company_Name,
    Energy_Consumption_MWh
FROM supply_chain  -- Correct table name
ORDER BY Energy_Consumption_MWh DESC
LIMIT 10;
'''

# Execute the query and fetch the results into a DataFrame
top_companies = pd.read_sql_query(query, conn)

# Display the top companies
print("Top 10 Companies with the Highest Energy Consumption:")
print(top_companies)

Top 10 Companies with the Highest Energy Consumption:
  Company_Name  Energy_Consumption_MWh
0     Facebook                  102000
1       Amazon                  101500
2       Google                  101000
3    Microsoft                  100500
4        Apple                  100000


#### 8. Customer Satisfaction vs. Supply Chain Agility

This query explores how customer satisfaction correlates with supply chain agility.

In [21]:
query_9 = '''
SELECT 
    Customer_Satisfaction,
    Supply_Chain_Agility
FROM supply_chain
ORDER BY Customer_Satisfaction DESC;
'''

# Execute the query and fetch the results into a DataFrame
customer_satisfaction_vs_agility = pd.read_sql_query(query_9, conn)

# Display the results
print("Customer Satisfaction vs. Supply Chain Agility:")
print(customer_satisfaction_vs_agility)

Customer Satisfaction vs. Supply Chain Agility:
   Customer_Satisfaction Supply_Chain_Agility
0                  91.00                 High
1                  90.00                 High
2                  90.00                 High
3                  89.00                 High
4                  88.00                 High


#### 9. Carbon Emissions and Use of Renewable Energy

This query examines the relationship between carbon emissions and the use of renewable energy across companies.

In [29]:
query_10 = '''
SELECT 
    Company_Name,
    Carbon_Emissions_kg_CO2e,
    Use_of_Renewable_Energy
FROM supply_chain
ORDER BY Carbon_Emissions_kg_CO2e DESC;
'''

# Execute the query and fetch the results into a DataFrame
carbon_emissions_and_renewable_energy = pd.read_sql_query(query_10, conn)

# Display the results
print("Carbon Emissions and Use of Renewable Energy:")
print(carbon_emissions_and_renewable_energy)

Carbon Emissions and Use of Renewable Energy:
  Company_Name  Carbon_Emissions_kg_CO2e  Use_of_Renewable_Energy
0     Facebook                    124000                       54
1       Amazon                    123000                       53
2       Google                    122000                       52
3    Microsoft                    121000                       51
4        Apple                    120000                       50


#### 10. Companies with High Environmental Impact Score

This query lists the companies that have the highest environmental impact score (e.g., greater than 75%).

In [27]:
query_11_adjusted = '''
SELECT 
    Company_Name,
    Environmental_Impact_Score
FROM supply_chain
WHERE Environmental_Impact_Score >= 70
ORDER BY Environmental_Impact_Score DESC;
'''

# Execute the adjusted query and fetch the results into a DataFrame
high_impact_companies_adjusted = pd.read_sql_query(query_11_adjusted, conn)

# Display the results
print("Companies with High Environmental Impact Score (Adjusted):")
print(high_impact_companies_adjusted)

Companies with High Environmental Impact Score (Adjusted):
  Company_Name  Environmental_Impact_Score
0    Microsoft                       75.00
1     Facebook                       75.00
2        Apple                       70.00
3       Google                       70.00
4       Amazon                       70.00


## Conclusion

This project analyzes and evaluates the supply chain management practices and sustainability efforts of top companies in the tech industry, focusing on their energy consumption, carbon emissions, use of renewable energy, green packaging, and overall environmental impact. Additionally, we examined the relationship between supply chain agility, customer satisfaction, and advanced technologies (such as AI, blockchain, and robotics). The findings offer valuable insights for businesses looking to improve their environmental impact and operational efficiency.

#### Key Insights:

1. **Supply Chain Practices**:
- Most companies in the dataset employ agile or lean supply chain practices, with Google, Amazon, Apple, Microsoft, and Facebook all utilizing these methods. These practices are known for enhancing supply chain flexibility, improving response times, and potentially boosting customer satisfaction.

2. **Customer Satisfaction and Lead Time**:
- There is a negative correlation (-0.4326) between lead time and customer satisfaction, suggesting that shorter lead times generally lead to higher customer satisfaction. Companies with very fast lead times (≤5 days) had high customer satisfaction, such as Google and Apple.
- Companies with moderate lead times (11-15 days) still maintained high levels of customer satisfaction, indicating that fast fulfillment may not always be the only factor influencing customer satisfaction.

3. **Use of Advanced Technologies**:
- Leading companies like Amazon, Apple, and Facebook utilize advanced technologies such as ERP, AI, blockchain, and robotics, contributing to their agile supply chain practices. These technologies improve operational efficiency and enable smarter decision-making in managing inventory and fulfillment.

4. **Environmental Impact and Sustainability**:
- The carbon emissions data reveals that Apple, Microsoft, Google, Amazon, and Facebook all produce significant carbon emissions, with Facebook having the highest emissions. However, all these companies are actively integrating renewable energy into their operations, though their carbon emissions remain high.
- The green packaging usage is relatively modest across the companies, with Apple and Microsoft showing slightly better results than the others.
- Despite these efforts, the project could not identify companies with both low carbon emissions and high green packaging usage, suggesting that further efforts are needed to balance these two environmental aspects.

5. **Sustainability Rankings**:
- Companies were ranked based on their sustainability scores, which integrate factors like carbon emissions, green packaging usage, and environmental impact scores. All companies in the dataset scored similarly (0.60), indicating that while they have made strides toward sustainability, there is still room for significant improvement.

6. **Energy Consumption**:
- Facebook has the highest energy consumption among the top companies, followed closely by Amazon and Google. These companies consume substantial amounts of energy, which suggests an area where they could improve energy efficiency and further reduce their environmental footprint.

7. **Environmental Impact Scores**:
- Companies like Microsoft, Facebook, Apple, and Google have relatively high environmental impact scores, which indicate that while they are taking some steps toward sustainability, their overall impact remains substantial. This highlights the need for continued innovation in reducing carbon emissions and enhancing sustainable practices across their operations.

### Why is this Important?

- **Environmental Sustainability**: The focus on energy consumption, carbon emissions, and renewable energy is critical for companies aiming to meet global sustainability goals, especially in light of climate change. Companies with a high environmental impact or inefficient energy use can benefit from insights that guide them to become more sustainable.

- **Corporate Responsibility**: Analyzing environmental scores and carbon footprints can help businesses make informed decisions about improving their sustainability practices, thus enhancing their reputation and meeting regulations related to climate action.

- **Supply Chain Performance**: By looking at supply chain agility and customer satisfaction, this project can help businesses improve efficiency and responsiveness, which could lead to both better business performance and customer loyalty.